sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/Hotspots/DynamicExecutionBase.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class DynamicExecutionBase<TSyntaxKind> : SonarDiagnosticAnalyzer
+        where TSyntaxKind : struct
+    {
+        protected const string DiagnosticId = "S1523";
+        protected const string MessageFormat = "Make sure that this dynamic injection or execution of code is safe.";
+
+        protected abstract InvocationTracker<TSyntaxKind> CreateInvocationTracker();
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            // Special case - Assembly.Load
+            var tracker1 = CreateInvocationTracker();
```

Comment:
Why do we need different tracker instances? It should work with only one I think...